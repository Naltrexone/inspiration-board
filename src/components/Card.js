import React from 'react';
import PropTypes from 'prop-types';

import './Card.css';

const Card = (props) => {

  console.log(typeof props.deleteCard);

  const {text, emoji } =  props.card;

  return(
    <div className="card">

      <div className="card__content">
        <div className="card__content-emoji">
          {emoji}
        </div>
        <div className="card__content-text">
          {text}
        </div>
        <button
          className="card__delete"
          onClick={props.deleteCard}>
          Delete
        </button>
      </div>
    </div>
  );
};

Card.propTypes = {
  card: PropTypes.object.isRequired,
  deleteCard: PropTypes.func.isRequired
};

export default Card;
