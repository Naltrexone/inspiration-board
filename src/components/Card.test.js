import React from 'react';
import ReactDOM from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  test('That it matches an existing snapshot', () => {
    const card = shallow(
      <Card
      id={1}
      text={'test message'}
      emoji={'test_emoji_code'}
      removeCardCallback={() => {}}
      />
    );

    expect(card).toMatchSnapshot();

    card.unmount();
  });
});
