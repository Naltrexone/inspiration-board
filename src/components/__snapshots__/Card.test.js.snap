exports[`Card That it matches an existing snapshot 1`] = `
<section
  className="card"
>
  <div
    className="card__content"
  >
    <p
      className="card__content-text"
    >
      test message
      <span
        className="card__content-emoji"
      />
    </p>
    <button
      onClick={[Function]}
    >
      delete
    </button>
  </div>
</section>
`;
